overlord/configstate/configcore/picfg.go

```diff
@@ -96,6 +97,10 @@ func piConfigFile(opts *fsOnlyContext) string {
 }
 
 func handlePiConfiguration(tr config.ConfGetter, opts *fsOnlyContext) error {
+	if release.OnClassic {
```

Comment:
we can probably do this by passing flags to the addConfig*Handler methods,  like coreOnlyConfig or classicOnlyConfig (though we don't have cases of the latter ATM)

Reply:
not sure what the flags should be, we should pick something so that we need to pass != 0 only in a few cases